test/fixtures/browser-app/app.js

```diff
@@ -7,4 +7,6 @@ var loopback = require('loopback');
 var boot = require('../../../');
 
 var app = module.exports = loopback();
-boot(app);
+app.start = function(done) {
+  boot(app, __dirname, done);
+};
```

Comment:
@davidcheung please revert this change and let's discuss how to achieve Raymond's requirement later, after this pull request is landed. That way the discussion can be focused on this single feature and does not get lost in this huge change-set.